ParkBlazerApp/src/pages/Reservations.tsx

```diff
@@ -0,0 +1,334 @@
+import { cardOutline } from 'ionicons/icons';
+import { useHistory } from "react-router-dom";
+import { bicycle, bus, car } from "ionicons/icons";
+import {
+    IonCol,
+    IonGrid,
+    IonRow,
+    IonButton,
+    IonText,
+    IonThumbnail,
+    IonCard,
+    IonCardContent,
+    IonItem,
+    IonLabel,
+    IonModal,
+    IonList,
+    IonContent,
+    
+} from "@ionic/react";
+import axios from "axios";
+import AuthService from "../utils/AuthService";
+import "./Reservations.css";
+import Navbar from "../components/navbar";
+import { parkingSpace, parkingspaces } from "../data/parkingSpaces";
+
+
+const OwnReservations: React.FC = () => {
+
+    interface Reservation {
+    reservation_id: number;
+    status: string;
+    start_date: string;
+    end_date: string;
+    timestamp: string;
+    private_parkingspot_id: number;
+    user_id: number;
+    }   
+
+
+    interface Parkingspot {
+    available_spaces: number;
+    city: string;
+    country: string;
+    description: string;
+    house_number: string;
+    image_url: string;
+    latitude: number;
+    longitude: number;
+    name: string;
+    parkingspot_id: number;
+    street: string;
+    type_bike: number;
+    type_car: number;
+    type_truck: number;
+    username: string;
+    zip: string;
+    distance?: number;
+    price_per_hour?: number;
+    availability_start_date?: string;
+    availability_end_date?: string;
+    }
+
+    interface CombinedData {
+    reservation: Reservation;
+    parkingspot: parkingSpace;
+    }
+
+    interface Invoice {
+        invoice_id: number;
+        amount: number;
+        payment_method: string;
+        payment_date: string;
+        reservation_id: number;
+        status: string;
+    }
+   
+    const [combinedData, setCombinedData] = useState<CombinedData[]>([]);
+    const [showModalInvoice, setShowModalInvoice] = useState<boolean>(false);
+    const [showModalInformation, setShowModalInformation] = useState<boolean>(false);
+    const [showModalImage, setShowModalImage] = useState<boolean>(false);
+    const [selectedInvoice, setSelectedInvoice] = useState<Invoice | null>(null);
+    const [currentParkingSpot, setcurrentParkingspot] = useState<parkingSpace | null>(null);
+    const [selectedImage, setSelectedImage] = useState('');
+    
+    function setError(message: any) {
+        throw new Error("Function not implemented.");
+    }
+
+    useEffect(() => {
+        const fetchUserData = async () => {
+            const token = AuthService.getToken();
+            try {
+                
+                const reservationsResponse = await axios.get('https://server-y2mz.onrender.com/api/user_reservations', {
+                    headers: {
+                        Authorization: `Bearer ${token}`
+                    }
+                });
+                
+                const reservations: Reservation[] = reservationsResponse.data;
+                const allparkingspaces: parkingSpace[] = parkingspaces;
```

Comment:
Hier fehlt der Aufruf ``await initParkingSpaces();`` um ``parkingspaces`` benutzten zu können
```suggestion
                await initParkingSpaces();
                const reservations: Reservation[] = reservationsResponse.data;
                const allparkingspaces: parkingSpace[] = parkingspaces;
```